internal/parser/parser.go

```diff
@@ -582,7 +582,25 @@ func (p *parser) parsePipeExpressionSuffix(expr *ast.Expression) func() bool {
 		}
 		// todo(jsternberg): this is not correct.
 		rhs := p.parseUnaryExpression()
-		call, _ := rhs.(*ast.CallExpression)
+		call, ok := rhs.(*ast.CallExpression)
+		if !ok && rhs != nil {
+			// We did not parse a call expression, but we still have something
+			// that was parsed so we need to pass over any errors from it
+			// and the location information so those remain present.
+			// We are losing the expression, so check for errors so that they
+			// are included in the output.
+			ast.Check(rhs)
+
+			// Copy the information to a blank call expression.
+			call = &ast.CallExpression{}
+			if loc := rhs.Location(); loc.IsValid() {
+				call.Loc = &loc
+			}
+			call.Errors = rhs.Errs()
+			call.Errors = append(call.Errors, ast.Error{
+				Msg: "pipe destination must be a function call",
```

Comment:
A little strange that the error checking is happening in two different places.  I guess that the RHS of `PipeExpression` is statically typed to be a `CallExpression` so you can't express.  Maybe making the RHS of `PipeExpression` be an `Expression` would help consolidate this?  The static type of `CallExpression` is helpful though, so I see arguments for doing it both ways.

Reply:
It largely depends on if we want to change the pipe expression to allow something else. If a pipe expression can output to something that's not a function, then we should change it to a function. Otherwise, I think this is fine and it doesn't involve changing the AST.

I think there might have been an idea to change the RHS to an expression though so this code would get removed and, if it still was applicable, would get moved to the semantic package if we did that.